sdk/java/pulumi/src/main/java/com/pulumi/automation/UpdateOptions.java

```diff
@@ -269,6 +285,19 @@ public B message(String message) {
             return (B) this;
         }
 
+        /**
+         * A list of resource URNs to exclude during the operation. Wildcards
+         * (*, **) are also supported.
+         *
+         * @param excludes the list of excluded resource URNs
+         * @return the builder
+         */
+        @SuppressWarnings("unchecked")
+        public B excludes(List<String> excludes) {
+            this.excludes = excludes;
+            return (B) this;
+        }
```

Comment:
Part of me wonders why `--*-dependents` isn't on `UpdateOptions`. I'm wondering if it's because of the bug/miss that you fixed when you joined, which is that `destroy` (or `refresh`?) didn't support `--target-dependents`? And so that was mirrored into the automation API. Given that we've now fixed this, I wonder if we can move `--target-dependents` up to `UpdateOptions`. It might be breaking -- not sure. Either way, not for this PR.

But what _is_ for this PR, is `--exclude-dependents`. It's not symmetric (and thus potentially ugly), but I think it's safe to put it on `UpdateOptions`, no? Might cut down the code a bit too?

Reply:
Related: this PR misses `RefreshOptions`, which omits `--target-dependents` (which we should fix) and thus (I think) confirms my theory

Reply:
Yeah, with this PR I was trying to follow the current structure - I agree that it seems safe to move the `--*-dependents` flag onto the `UpdateOptions`, but I'd equally be happy to move both `--*-dependents` flags to the `UpdateOptions` in a follow-up? .NET works in exactly the same way and I made a [near-identical PR](https://github.com/pulumi/pulumi-dotnet/pull/582), so I'm happy to do a follow-up for each one?

Reply:
I think moving both _might_ be a breaking change (not sure), so I was only thinking just "do this one right" but am happy either way.